src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -90,7 +90,32 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		@IWorkspaceConfigurationService private _workspaceConfigurationService: IWorkspaceConfigurationService,
 		@IClipboardService private _clipboardService: IClipboardService,
 		@ICommandService private _commandService: ICommandService
-	) { }
+	) {
+		this.initializeConnectionProviders();
+	}
+
+	private initializeConnectionProviders() {
+		this.setConnectionProviders();
+		if (this._capabilitiesService) {
+			this._capabilitiesService.onCapabilitiesRegistered(() => {
+				this.setConnectionProviders();
+				if (this._connectionDialog) {
+					this._connectionDialog.updateConnectionProviders(this._providerTypes, this._providerNameToDisplayNameMap);
+				}
+			});
+		}
+	}
+
+	private setConnectionProviders() {
+		if (this._capabilitiesService) {
+			this._providerTypes = [];
+			this._providerNameToDisplayNameMap = {};
+			entries(this._capabilitiesService.providers).forEach(p => {
+				this._providerTypes.push(p[1].connection.displayName);
```

Comment:
what does p[1] represent?

Reply:
this is actually existing code in the same file, just got moved here by me. I didn't really look into the details here. seems the provider's property is organized this way.